lib/private/Files/Cache/Propagator.php

```diff
@@ -116,18 +117,18 @@ public function propagateChange($internalPath, $time, $sizeDifference = 0) {
 
 			if ($this->storage->instanceOfStorage(Encryption::class)) {
 				// in case of encryption being enabled after some files are already uploaded, some entries will have an unencrypted_size of 0 and a non-zero size
+				$eq = $builder->expr()->eq('unencrypted_size', $builder->expr()->literal(0, IQueryBuilder::PARAM_INT));
+				$sizeColumn = $builder->getColumnName('size');
+				$unencryptedSizeColumn = $builder->getColumnName('unencrypted_size');
 				$builder->set('unencrypted_size', $builder->func()->greatest(
 					$builder->func()->add(
-						$builder->func()->case([
-							['when' => $builder->expr()->eq('unencrypted_size', $builder->expr()->literal(0, IQueryBuilder::PARAM_INT)), 'then' => 'size']
-						], 'unencrypted_size'),
+						new QueryFunction("CASE WHEN $eq THEN $unencryptedSizeColumn ELSE $sizeColumn END"),
```

Comment:
```suggestion
						$builder->createFunction("CASE WHEN $eq THEN $unencryptedSizeColumn ELSE $sizeColumn END"),
```
So you don't need to use private API directly